File: api/src/main/java/net/blueshell/api/validation/survey/ValidAnswerList.java
package net.blueshell.api.validation.survey;

import jakarta.validation.Constraint;
import jakarta.validation.Payload;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.FIELD})
@Retention(RetentionPolicy.RUNTIME)
@Constraint(validatedBy = ValidAnswerListValidator.class)
public @interface ValidAnswerList {
    String message() default "Invalid list of answers";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};
}

File: api/src/main/kotlin/net/blueshell/api/survey/validation/ValidAnswerList.kt
package net.blueshell.api.survey.validation

import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass

@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
@Constraint(validatedBy = [ValidAnswerListValidator::class])
annotation class ValidAnswerList(
    val message: String = "Invalid list of answers",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = []
)

